(/authentication/social-connections/discord) update guide (#1600)
clerk/clerk-docs: docs/authentication/social-connections/discord.mdx

description: Learn how to allow users to sign up and sign in to your Clerk app with their Discord account.
---

<TutorialHero
  beforeYouStart={[
    {
      title: "A Clerk app is required.",
      link: "/docs/quickstarts/setup-clerk",
      icon: "clerk",
    },
    {
      title: "A Discord account is required.",
      link: "https://discord.com/register",
      icon: "user-circle",
    }
  ]}
>
  - Use Discord to authenticate users with OAuth
</TutorialHero>

Enabling OAuth with [**Discord**](https://discord.com/developers/docs/topics/oauth2) allows your users to sign up and sign in to your Clerk application with their Discord account.

## Configure for your development instance

For _development instances_, Clerk uses preconfigured shared OAuth credentials and redirect URIs—no other configuration is needed.

1. In the Clerk Dashboard, navigate to the [**SSO Connections**](https://dashboard.clerk.com/last-active?path=user-authentication/sso-connections) page.
1. Select **Add connection** and select **For all users**.
1. In the **Choose provider** dropdown, select **Discord**.
1. Select **Add connection**.

## Configure for your production instance

> In _production instances_, you must provide custom credentials, which involves generating your own **Client ID** and **Client Secret** using your Microsoft Entra ID account.

Reviewer: ```suggestion
In _production instances_, you must provide custom credentials, which involves generating your own **Client ID** and **Client Secret** using your Discord account.
```

> To make the setup process easier, it's recommended to keep two browser tabs open: one for your [Clerk Dashboard](https://dashboard.clerk.com/last-active?path=user-authentication/sso-connections) and one for your [Discord Developers](https://discord.com/developers/applications) page.

Reviewer: ```suggestion
To make the setup process easier, it's recommended to keep two browser tabs open: one for your [Clerk Dashboard](https://dashboard.clerk.com/last-active?path=user-authentication/sso-connections) and one for your [Discord Developers Portal](https://discord.com/developers/applications).
```

We can call it what they call it!
<img width="559" alt="Screenshot 2024-10-09 at 10 30 28" src="https://github.com/user-attachments/assets/13b9bd52-d8a0-4a21-b0bb-bf1ae75ece47">

<Steps>
  ### Enable Discord as a social connection in Clerk

>   1. Navigate to the [Clerk Dashboard](https://dashboard.clerk.com/last-active?path=user-authentication/sso-connections).
>   1. In the top navigation, select **Configure**. Then in the sidebar, select **SSO Connections**.

Reviewer: ```suggestion
  1. In the Clerk Dashboard, navigate to the [**SSO Connections**](https://dashboard.clerk.com/last-active?path=user-authentication/sso-connections) page.
```

>   1. Select the **Add connection** button, and select **For all users**.

Reviewer: ```suggestion
  1. Select **Add connection** and select **For all users**.
```

  1. In the **Choose provider** dropdown, select **Discord**.
  1. Ensure that both **Enable for sign-up and sign-in** and **Use custom credentials** are toggled on.

>   1. Save the **Authorized redirect URI** somewhere secure.
>   1. Keep the modal and page open throughout the guide.

Reviewer: ```suggestion
  1. Save the **Authorized redirect URI** somewhere secure. Keep the modal and page open throughout the guide.
```

  ### Create a Discord Developer app

>   1. On a separate page, go to the [Discord Developers](https://discord.com/developers/applications) page and sign in to create a new Discord app.

Reviewer: ```suggestion
  1. On a separate page, go to the [Discord Developers Portal](https://discord.com/developers/applications) and sign in.
```

  1. In the top-right corner, select **New Application**.
  1. Complete the required fields and select **Create**. You'll be redirected to the **General Information** page.
  1. In the sidebar, select **OAuth2**.

>   1. Under **Redirects**, select **Add Redirect**. Paste the **Authorized Redirect URI** you saved from the Clerk Dashboard. Then, select **Save Changes**. You may need to tap anywhere on the screen for the button to appear. Keep this page open.

Reviewer: let's break these steps up.

```suggestion
  1. Under **Redirects**, select **Add Redirect**. Paste the **Authorized Redirect URI** you saved from the Clerk Dashboard. 
  1. Select **Save Changes**. You may need to tap anywhere on the screen for the button to appear. Keep this page open.
```

>   1. In the next section, you will connect Discord to Clerk using your **Client ID** and **Client Secret**. Keep this page open for the next step.

Reviewer: ```suggestion
  1. Keep this page open.
```

  ### Set the Client ID and Client Secret in your Clerk Dashboard

>   1. In your Discord Developers dashboard, copy the **Client ID** and **Client Secret**. If you don't see the **Copy** button under the **Client Secret**, select **Reset Secret** to generate a new one.

Reviewer: ```suggestion
  1. In your Discord Developers Portal, copy the **Client ID** and **Client Secret**. If you don't see the **Copy** button under the **Client Secret**, select **Reset Secret** to generate a new one.
```

>   1. In your Discord application's **OAuth2** page, copy the **Client ID**.
>   1. Navigate back to your **Clerk Dashboard** page. In the **Add connection for all users** modal under the **Configure Discord**, paste the string in **Client ID**.
>   1. Navigate back to your Discord application's **OAuth2** page. Select  **Copy** under **Client Secret**. If you do not see the **Copy** button, select **Reset Secret** to generate a new one. Be sure to copy the string as it will not be shown again.
>   1. Navigate back to your **Clerk Dashboard** page. Paste the string in **Client Secret** and select **Add connection**. Once the configurations are saved, you should see a green **`Used for sign-in`** label next to Discord in the **Provider** list.

Reviewer: ```suggestion
  1. In your Discord Developers dashboard, copy the **Client ID** and **Client Secret**. If you don't see the **Copy** button under the **Client Secret**, select **Reset Secret** to generate a new one.
  1. Navigate back to your **Clerk Dashboard** where the modal should still be open, and paste these values into the respective fields.
  1. Select **Add connection**.
  
  > [!NOTE]
  > If the modal or page is not still open, navigate to the [Clerk Dashboard](https://dashboard.clerk.com/last-active?path=user-authentication/sso-connections). In the top navigation, select **Configure**. Then in the sidebar, select **SSO Connections**. Select the **Spotify** connection. Under **Use custom credentials**, you can paste the **Client ID** and **Client Secret** into their respective fields.
```

  > [!NOTE]

>   > If the modal or page is not still open, navigate to the [Clerk Dashboard](https://dashboard.clerk.com/last-active?path=user-authentication/sso-connections). In the top navigation, select **Configure**. Then in the sidebar, select **SSO Connections**. Select the **Discord** connection. Under **Use custom credentials**, you can paste the **Client ID** and **Client Secret** into their respective fields.

Reviewer: ```suggestion
  > If the modal or page is not still open, navigate to the [**SSO Connections**](https://dashboard.clerk.com/last-active?path=user-authentication/sso-connections) page in the Clerk Dashboard. Select the **Discord** connection. Under **Use custom credentials**, you can paste the **Client ID** and **Client Secret** into their respective fields.
```

  The simplest way to test your OAuth setup is by visiting your Clerk application's [Account Portal](/docs/customization/account-portal/overview), which is available out-of-the-box for all Clerk applications.

  1. In the navigation sidebar of your **Clerk Dashboard**, select [**Account Portal**](https://dashboard.clerk.com/last-active?path=account-portal).
  1. Next to the **Sign-in** URL, select **Visit**. The URL should resemble:
     - **For development**: `https://your-domain.accounts.dev/sign-in`
     - **For production**: `https://accounts.your-domain.com/sign-in`
  1. On the sign-in page, you should see **Discord** as an option. Use it to sign in with your Discord account.
</Steps>

> ### Finished 🎉
> 
> Congratulations! Your users can now sign in and sign up for your Clerk application with their Discord account.

Reviewer: ```suggestion
```
